fdbrpc/genericactors.actor.h

```diff
@@ -70,6 +72,100 @@ Future<REPLY_TYPE(Req)> retryBrokenPromise(RequestStream<Req> to, Req request, T
+                                                         Req request,
+                                                         Hostname hostname,
+                                                         WellKnownEndpoints token) {
+	// A wrapper of tryGetReply(request), except that the request is sent to an address resolved from a hostname.
+	// If resolving fails, return lookup_failed().
+	// Otherwise, return tryGetReply(request).
+	try {
+		hostname.resolveBlocking();
+	} catch (...) {
+		return ErrorOr<REPLY_TYPE(Req)>(lookup_failed());
+	}
+	Optional<NetworkAddress> address = hostname.resolvedAddress;
+	*to = RequestStream<Req>(Endpoint::wellKnown({ address.get() }, token));
+	return to->tryGetReply(request);
+}
+
+template <class Req>
+Future<ErrorOr<REPLY_TYPE(Req)>> tryGetReplyFromHostname(RequestStream<Req>* to,
+                                                         Req request,
+                                                         Hostname hostname,
+                                                         WellKnownEndpoints token,
+                                                         TaskPriority taskID) {
+	// A wrapper of tryGetReply(request), except that the request is sent to an address resolved from a hostname.
+	// If resolving fails, return lookup_failed().
+	// Otherwise, return tryGetReply(request).
+	try {
+		hostname.resolveBlocking();
+	} catch (...) {
+		return ErrorOr<REPLY_TYPE(Req)>(lookup_failed());
+	}
+	Optional<NetworkAddress> address = hostname.resolvedAddress;
+	*to = RequestStream<Req>(Endpoint::wellKnown({ address.get() }, token));
+	return to->tryGetReply(request, taskID);
+}
+
+ACTOR template <class Req>
+Future<REPLY_TYPE(Req)> retryGetReplyFromHostname(RequestStream<Req>* to,
+                                                  Req request,
+                                                  Hostname hostname,
+                                                  WellKnownEndpoints token) {
+	// Like to.getReply(request), except that a request_maybe_delivered exception results in re-resolving the hostname.
+	// Suitable for use with hostname, where RequestStream is NOT initialized yet.
+	// Not normally useful for endpoints initialized with NetworkAddress.
+	loop {
+		wait(hostname.resolve());
+		state Optional<NetworkAddress> address = hostname.resolvedAddress;
+		*to = RequestStream<Req>(Endpoint::wellKnown({ address.get() }, token));
+		ErrorOr<REPLY_TYPE(Req)> reply = wait(to->tryGetReply(request));
+		if (reply.isError()) {
+			resetReply(request);
+			if (reply.getError().code() == error_code_request_maybe_delivered) {
+				// Connection failure.
+				hostname.resetToUnresolved();
+				INetworkConnections::net()->removeCachedDNS(hostname.host, hostname.service);
+			} else {
+				throw reply.getError();
+			}
+		} else {
+			return reply.get();
+		}
+	}
+}
+
+ACTOR template <class Req>
+Future<REPLY_TYPE(Req)> retryGetReplyFromHostname(RequestStream<Req>* to,
```

Comment:
Can we combine this function with the above function, with `TaskPriority taskID` being an optional parameter?